index.js

```diff
@@ -3,76 +3,84 @@ const sortObjectKeys = require('sort-object-keys');
 const detectIndent = require('detect-indent');
 const glob = require('glob');
 
-const sortOrder = [
-  'name',
-  'version',
-  'private',
-  'description',
-  'keywords',
-  'homepage',
-  'bugs',
-  'repository',
-  'funding',
-  'license',
-  'author',
-  'contributors',
-  'files',
-  'sideEffects',
-  'type',
-  'exports',
-  'main',
-  'umd:main',
-  'jsdelivr',
-  'unpkg',
-  'module',
-  'source',
-  'jsnext:main',
-  'browser',
-  'types',
-  'typings',
-  'style',
-  'example',
-  'examplestyle',
-  'assets',
-  'bin',
-  'man',
-  'directories',
-  'workspaces',
-  'scripts',
-  'betterScripts',
-  'husky',
-  'pre-commit',
-  'commitlint',
-  'lint-staged',
-  'config',
-  'nodemonConfig',
-  'browserify',
-  'babel',
-  'browserslist',
-  'xo',
-  'prettier',
-  'eslintConfig',
-  'eslintIgnore',
-  'stylelint',
-  'ava',
-  'jest',
-  'mocha',
-  'nyc',
-  'dependencies',
-  'devDependencies',
-  'peerDependencies',
-  'bundledDependencies',
-  'bundleDependencies',
-  'optionalDependencies',
-  'flat',
-  'resolutions',
-  'engines',
-  'engineStrict',
-  'os',
-  'cpu',
-  'preferGlobal',
-  'publishConfig'
+// field.key{string}: field name
+// field.sortSubKey{boolean}: sort field subKey
+// field.sortList{string[]}: key order array
+// field.unique{boolean}: unique array values
+// field.sortScripts{boolean}: sort field as scripts
+const fields = [
+  {key: 'name', sortSubKey: false},
+  {key: 'version', sortSubKey: false},
+  {key: 'private', sortSubKey: true},
+  {key: 'description', sortSubKey: false},
+  {key: 'keywords', sortSubKey: true, unique: true},
+  {key: 'homepage', sortSubKey: true},
+  {key: 'bugs', sortSubKey: true, sortList: ['url', 'email']},
+  {key: 'repository', sortSubKey: true, sortList: ['type', 'url']},
+  {key: 'funding', sortSubKey: true, sortList: ['type', 'url']},
+  {key: 'license', sortSubKey: true, sortList: ['type', 'url']},
+  {key: 'author', sortSubKey: true, sortList: ['name', 'email', 'url']},
```

Comment:
Do you think this is a little complicated?

I was trying to make this simplier like this


```js
{
  'name',
  'version',
  {key: 'homepage'},
  {key: 'bugs', sortList: ['url', 'email']}
}
```

But, seems not very clear that `homepage` and `bugs` will sort sub key, any suggestion?